src/angular/components/password-generator.component.ts

```diff
@@ -31,13 +31,17 @@ export class PasswordGeneratorComponent implements OnInit {
         this.options = optionsResponse[0];
         this.enforcedPolicyOptions = optionsResponse[1];
         this.policyInEffect = this.enforcedPolicyOptions != null && (
+            this.enforcedPolicyOptions.defaultType !== '' ||
```

Comment:
Maybe we add an `inEffect(): boolean` function within this object that we just call that does all of this? Seems this is getting a bit out of hand.

Reply:
Added a helper method to the enforced options object.